File: src/main/kotlin/no/nav/arbeidsgiver/min_side/services/digisyfo/DigisyfoRepositoryStub.java
package no.nav.arbeidsgiver.min_side.services.digisyfo;


import org.apache.commons.lang3.tuple.ImmutablePair;
import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Repository;

import java.util.List;

@Profile({"local", "labs"})
@Repository
public class DigisyfoRepositoryStub implements DigisyfoRepository {
    @Override
    public List<Virksomhetsinfo> virksomheterOgSykmeldte(String nærmestelederFnr) {
        return List.of(new Virksomhetsinfo("910825526", 4));
    }
    @Override
    public void processNærmesteLederEvent(NarmesteLederHendelse hendelse) {
    }

    @Override
    public void processSykmeldingEvent(List<ImmutablePair<String, SykmeldingHendelse>> records) {
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/services/digisyfo/DigisyfoRepositoryStub.kt
package no.nav.arbeidsgiver.min_side.services.digisyfo

import no.nav.arbeidsgiver.min_side.services.digisyfo.DigisyfoRepository.Virksomhetsinfo
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Repository

@Profile("local", "labs")
@Repository
class DigisyfoRepositoryStub : DigisyfoRepository {
    override fun virksomheterOgSykmeldte(nærmestelederFnr: String): List<Virksomhetsinfo> {
        return listOf(Virksomhetsinfo("910825526", 4))
    }

    override fun processNærmesteLederEvent(hendelse: NarmesteLederHendelse) {}
    override fun processSykmeldingEvent(records: List<Pair<String?, SykmeldingHendelse?>>) {}
}